### PR TITLE
feat: add support for icons and loading state on all buttons

### PR DESCRIPTION
**Pull request by jenseo**

### CODE REVIEW

File: src/components/controls/buttons/PrimaryButton/PrimaryButton.tsx [hunk 1]
```diff
@@ -1,21 +1,46 @@
...
+          {children}
+          <AnimatePresence>
+            {(icon || isLoading) && (
+              <IconWrapper
+                initial={{ marginLeft: 0, opacity: 0, width: 0 }}
+                animate={{
+                  marginLeft: isLoading ? "40px" : "16px",
+                  opacity: 1,
+                  width: "auto",
+                }}
```
Comment (filiptammergard) [inline src/components/controls/buttons/PrimaryButton/PrimaryButton.tsx:33]: I know the loading state in Figma makes the button wider, but I wonder if that's really intended. We should talk this through with Rui 🙂

File: src/components/controls/buttons/PrimaryButton/PrimaryButton.tsx [hunk 2]
```diff
@@ -1,21 +1,46 @@
...
 import { ButtonHTMLAttributes, ElementType, forwardRef, ReactNode } from "react"
 import { useWidthFromColumns } from "../../../../hooks/useWidthFromColumns"
+import { Loader } from "../../../feedback/Loader/Loader"
 import { BaseButton } from "../BaseButton/BaseButton"
 
 export interface PrimaryButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
   as?: ElementType
   children: ReactNode
   columns?: number | number[]
+  icon?: ReactNode
```
Comment (filiptammergard) [inline src/components/controls/buttons/PrimaryButton/PrimaryButton.tsx:12]: I really like how [Mantine](https://mantine.dev/core/button/#loading-state) and [Chakra](https://chakra-ui.com/docs/components/form/button#button-with-icon) handle this. I think we should name this prop `rightIcon`, but have the possibility to introduce `leftIcon` in the future`.

File: src/components/controls/buttons/PrimaryButton/PrimaryButton.tsx (hunk 1 above)
Comment (jenseo) [reply src/components/controls/buttons/PrimaryButton/PrimaryButton.tsx:33]: Hmm, yeah, let's talk with him.

Comment (filiptammergard) [reply src/components/controls/buttons/PrimaryButton/PrimaryButton.tsx:33]: I talked to him quickly last week 🙂 

image

Comment (jenseo) [reply src/components/controls/buttons/PrimaryButton/PrimaryButton.tsx:33]: Ah, great, was just about to write him! :)

File: src/components/controls/buttons/PrimaryButton/PrimaryButton.tsx [hunk 3]
```diff
@@ -50,3 +75,12 @@ const StyledBaseButton = styled(BaseButton)<StyledBaseButtonProps>`
...
   }
 `
+const IconWrapper = styled(motion.div)<{ isLoading: boolean }>`
+  color: ${({ theme }) => theme.colors.buttons.icon.primary};
+  width: ${({ theme }) => 3 * theme.spacer}px;
+`
+const ContentWrapper = styled.div`
+  display: flex;
+  justify-content: center;
+  align-items: center;
```
[the remark this answers is not in the document]
Comment (jenseo) [reply src/components/controls/buttons/PrimaryButton/PrimaryButton.tsx:85]: I guess so yes, I looked at how it had been done in IconWithButton to align as much as possible with how you had done it previously. But makes sense to get rid of some wrappers :)

File: src/components/controls/buttons/BaseButton/BaseButtonIcon.tsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+const Wrapper = styled.div`
+  color: ${({ theme }) => theme.colors.buttons.icon.primary};
+  margin-left: ${({ theme }) => 2 * theme.spacer}px;
+`
+
+const IconWrapper = styled.div`
+  width: ${({ theme }) => 3 * theme.spacer}px;
+  height: ${({ theme }) => 3 * theme.spacer}px;
+`
```
Comment (filiptammergard) [inline src/components/controls/buttons/BaseButton/BaseButtonIcon.tsx:23]: I added this to make sure the icon is in the same size as the loader, to make sure the button is not resized when it shouldn't 🙂 Sounds good?

File: src/components/controls/buttons/PrimaryButton/PrimaryButton.tsx [hunk 5]
```diff
@@ -2,20 +2,31 @@ import styled from "@emotion/styled"
...
   children: ReactNode
   columns?: number | number[]
+  isLoading?: boolean
   isFullWidth?: boolean
+  rightIcon?: ReactNode
 }
 
 export const PrimaryButton = forwardRef<HTMLButtonElement, PrimaryButtonProps>(
-  ({ children, isFullWidth = false, ...props }, ref) => {
+  ({ children, isLoading = false, rightIcon, isFullWidth = false, ...props }, ref) => {
```
Comment (filiptammergard) [inline src/components/controls/buttons/PrimaryButton/PrimaryButton.tsx:17]: I think we can think about it a bit more, but we probably want to handle the buttons being disabled when loading is true. But this is good for now!

**Review by filiptammergard — APPROVED** [review]
Nice!

File: src/components/controls/buttons/BaseButton/BaseButtonIcon.tsx (hunk 4 above)
Comment (jenseo) [reply src/components/controls/buttons/BaseButton/BaseButtonIcon.tsx:23]: Sounds awesome, I wanted to bounce that with you, forgot about it :D

File: src/components/controls/buttons/PrimaryButton/PrimaryButton.tsx (hunk 5 above)
Comment (jenseo) [reply src/components/controls/buttons/PrimaryButton/PrimaryButton.tsx:17]: Agreed! Let's do that next iteration

**Pull request merged**